### PR TITLE
Add SingaporeProvider

### PR DESCRIPTION
**Pull request by YipingRuan**

### CODE REVIEW

File: Src/Nager.Date/PublicHolidays/SingaporeProvider.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+            return items.OrderBy(o => o.Date);
+        }
+
+        ///<inheritdoc/>
+        public IEnumerable<string> GetSources()
+        {
+            return new string[]
+            {
+                "https://www.mom.gov.sg/employment-practices/public-holidays#Year-2022"
```
Comment (samtrion) [inline Src/Nager.Date/PublicHolidays/SingaporeProvider.cs:56]: We try to at least support the last years and the next x years.

**Review by samtrion — CHANGES_REQUESTED** [review]

File: Src/Nager.Date.UnitTest/Country/SingaporeTest.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    [TestClass]
+    public class SingaporeTest
+    {
+        [TestMethod]
+        public void TestSingapore()
+        {
+            var publicHolidays = DateSystem.GetPublicHolidays(2022, CountryCode.SG).ToArray();
+
+            Assert.AreEqual("New Year’s Day", publicHolidays[0].Name);
```
Comment (samtrion) [inline Src/Nager.Date.UnitTest/Country/SingaporeTest.cs:16]: Please add more tests, especially for the years with optional holidays like 2022

**Review by samtrion — CHANGES_REQUESTED** [review]

File: Src/Nager.Date/PublicHolidays/SingaporeProvider.cs (hunk 1 above)
Comment (YipingRuan) [reply Src/Nager.Date/PublicHolidays/SingaporeProvider.cs:56]: Agree. Would be better is this is mentioned in the contribution guide.

**Comment by YipingRuan** [comment]
https://github.com/nager/Nager.Date/issues/275

**Comment by tinohager** [comment]
@YipingRuan Good Friday is a calculated holiday available via [ICatholicProvider](https://github.com/nager/Nager.Date/blob/master/Src/Nager.Date/PublicHolidays/IsleOfManProvider.cs)

File: Src/Nager.Date/PublicHolidays/SingaporeProvider.cs [hunk 3]
```diff
@@ -32,45 +37,53 @@ public IEnumerable<PublicHoliday> Get(int year)
...
+            // Good Friday
+            items.Add(this._catholicProvider.GoodFriday("Good Friday", year, countryCode));
+
+            // Chinese New Year, 2 days
+            if (year > 1901 && year < 2100)
+            {
+                //LunisolarCalendar .net implementation only valid are between 1901 and 2100, inclusive.
+                //https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Globalization/ChineseLunisolarCalendar.cs
+                //https://stackoverflow.com/questions/30719176/algorithm-to-find-the-gregorian-date-of-the-chinese-new-year-of-a-certain-gregor
+                var chineseCalendar = new ChineseLunisolarCalendar();
```
[the remark this answers is not in the document]
Comment (YipingRuan) [reply Src/Nager.Date/PublicHolidays/SingaporeProvider.cs:54]: Can have more hint? I copied the code from https://github.com/nager/Nager.Date/blob/fc6de194942b1408439c4078089fa6d269cf1db7/Src/Nager.Date/PublicHolidays/SurinameProvider.cs#L59

**Review by tinohager — APPROVED** [review]

**Pull request merged**